### ISSUE TITLE
create nested json tree structure

### ISSUE THREAD

**Issue by doidata** [mhaemmerle/excel-to-json#20]
Hello and I assume this is not an issue but rather a  request, thanks and sorry for this.
I try to create nested json array structures as an input file for d3js visualisation from excel files. I can create ONE array nest with the current excel-to-json succesfully, but cannot figure out how to create deeper array "children". The blueprint for the json would be the (infamous) "flare.json" structure which is the showcase for most d3js treemaps etc.:
https://gist.github.com/mbostock/1093025
I have all the freedom to create the excel structur,e as I am just building the data from ground up. Is there a way to create this already now, or are the maybe some easy changes to your code to make this possible?
dankeschoen aus wien, karl.

**Comment by adunning** [comment]
Did you eventually find a way of accomplishing this? I'm in a similar situation.

(Thanks, by the way, to everyone who has worked on this tool; it's fantastic.)

**Comment by doidata** [comment]
Yes I did find a way, but not with this tool. After coming up with a structure and notation in this thread there was no further response. As I am not a real coder I had to hire someone via elancer and had him program an excel vba script for me, which works perfectly for the output to d3js json. If you can work with excel vba in your environment I can send you the vba script for free, as I explicitely asked the programmer to give me the rights for public domain usage.
Greetings Karl.

**Comment by eproxus** [comment]
@doidata If you don't mind, maybe link to the code from this issue? It would help other people having the same problem.

We started looking at supporting this feature in excel-to-json as well, but we haven't had time to properly implement something yet. This issue should stay open until we do.

**Comment by MasterSquishy** [comment]
@doidata Could you please link to the vba script code for this solution?

**Comment by doidata** [comment]
Hello yes i will
Just out if touch with this topic for some months now i have to find time to search my harddrive.... Over weekend maybe
Cheers karl

Chanelle <notifications@github.com> schrieb am 29.09.2015:
>@doidata Could you please link to the vba script code for this
>solution?
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/mhaemmerle/excel-to-json/issues/20#issuecomment-144095745

-- Diese Nachricht wurde mit K-@ Mail gesendet.

**Comment by MasterSquishy** [comment]
Thank you, much appreciated!

*Chanelle D. Green*

B.S. Computer Science
*Spelman College '10*

B.S.E. Industrial & Operations Engineering
*University of Michigan '13*

Campus Ambassador
*Electronic Arts (EA)*

**Comment by doidata** [comment]
Hello so here is the excel sheet with embedde vba code (ExcelJSON.xlsm),
there are some rules for correct usage which you find in the comments below
written by the programmer. There is also a sample data xml sheet for
testing.
I wish you success with your project, greetings, Karl.

Few changes:
- root parent can be in any row (not only first line)
- nodes can have attributes (optional)
- strings (characters like quotes) are encoded to json standard (e.g. an"na
will be converted to an\"na but 3djs decodes it back to an"na)
- any number type (integer, decimal, currency) is converted to json number
- true and false are converted to json boolean
- null is converted to json null

Rules:
- there can be only one 'root' node, if there are 2 or more roots script
picks last one
- parent id must be unique, (e.g. when you add two nodes with id "node_a"
script will return empty string)

**Comment by chanellegreen** [comment]
@doidata Thank you!...but there's no attachment :-)

**Comment by doidata** [comment]
Ok it seems that github mail drops the attachements? Please mail me directly at doidata@gmail.com
Greetings karl.

chanellegreen <notifications@github.com> schrieb am 10.10.2015:
>@doidata Thank you!...but there's no attachment :-)
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/mhaemmerle/excel-to-json/issues/20#issuecomment-147015962

-- Diese Nachricht wurde mit K-@ Mail gesendet.

**Comment by JawadJaber** [comment]
Hi, can you email me the excel to jawadsab@gmail.com.

Many thanks.

**Comment by csalvado** [comment]
Hello Doidata

Is it possible to share the VBA script?  I might be able to crack the children nesting to Json from the excel.
im going to send an email to you directily.
Many thanks

**Comment by KunalShahare** [comment]
Hello Doidata,

Can you please share the VBA script for this issue.
I am stuck with similar issue from last 2 weeks , it would be really helpful if you share the vba script for this solution.

Many thanks.

**Comment by anicu24** [comment]
Hello @doidata,

Can you please send me also the VBA script? 
I have an excel table with some filters and would need to convert it to a JSON file as tree data structure.

I am strugling to create the VBA script.

Thank you very much
